zmon_aws_agent/aws.py

```diff
@@ -298,23 +300,24 @@ def get_running_apps(region, existing_entities=None):
 
             result.append(ins)
 
-        if now.hour == 21:
-            imgs = []
-            try:
-                imgs = aws_client.describe_images(ImageIds=list(images))['Images']
-                for i in result:
-                    if 'image_id' not in i:
-                        continue
-                    for img in imgs:
-                        if img['ImageId'] == i['image_id']:
-                            i['image_name'] = img['Name'] if 'Name' in img else 'UNKNOWN'
-                            if 'CreationDate' in img:
-                                i['image_date'] = img['CreationDate'].replace('Z', '+00:00')
-                            else:
-                                i['image_date'] = '1970-01-01T00:00:00.000+00:00'
-                            break
-            except:
-                pass
+        logger.info('Fetching {} images: {}'.format(len(images), list(images)))
+        imgs = []
+        try:
+            imgs = aws_client.describe_images(ImageIds=list(images))['Images']
+            logger.info('Got {} image descriptions'.format(len(imgs)))
+            for i in result:
+                if 'image' not in i:
+                    continue
+                for img in imgs:
+                    if img['ImageId'] == i['image']['id']:
+                        i['image']['name'] = img['Name'] if 'Name' in img else 'UNKNOWN'
+                        if 'CreationDate' in img:
+                            i['image']['date'] = img['CreationDate'].replace('Z', '+00:00')
+                        else:
+                            i['image']['date'] = '1970-01-01T00:00:00.000+00:00'
```

Comment:
I am just wondering whether this will always trigger an alert? Is it better to keep it empty string in the unlikely case of CreationDate does not exist?